maat-court-data-api/src/test/java/gov/uk/courtdata/service/QueueMessageLogServiceTest.java

```diff
@@ -0,0 +1,189 @@
+package gov.uk.courtdata.service;
+
+
+import gov.uk.courtdata.entity.QueueMessageLogEntity;
+import gov.uk.courtdata.enums.JurisdictionType;
+import gov.uk.courtdata.enums.QueueMessageType;
+import gov.uk.courtdata.repository.QueueMessageLogRepository;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.runner.RunWith;
+import org.mockito.*;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueueMessageLogServiceTest {
+
+    @InjectMocks
+    public QueueMessageLogService queueMessageLogService;
+    @Spy
+    public QueueMessageLogRepository queueMessageLogRepository;
+    @Captor
+    private ArgumentCaptor<QueueMessageLogEntity> queueMessageCaptor;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+
+    @Test
+    public void whenLinkMessage_CheckLogEntryCreated() {
+
+        final Integer maatId = 1000;
+
+        queueMessageLogService.createLog(QueueMessageType.LINK, newQueueMessage(maatId));
+
+        verify(queueMessageLogRepository, times(1)).save(queueMessageCaptor.capture());
+
+        QueueMessageLogEntity savedQueueMsg = queueMessageCaptor.getValue();
+
+        assertAll("linkMessage",
+                () -> assertNotNull(savedQueueMsg),
+                () -> assertNotNull(savedQueueMsg.getMaatId()),
+                () -> assertNotNull(savedQueueMsg.getTransactionUUID()),
+                () -> assertNotNull(savedQueueMsg.getType()),
+                () -> assertNotNull(savedQueueMsg.getMessage()),
+                () -> assertNotNull(savedQueueMsg.getCreatedTime()),
+                () -> assertEquals(savedQueueMsg.getMaatId(), maatId),
+                () -> assertEquals(savedQueueMsg.getType(), QueueMessageType.LINK.name())
+
+        );
+    }
+
+    @Test
+    public void whenUnLinkMessage_CheckLogEntryCreated() {
+
+        final Integer maatId = 1000;
+
+        queueMessageLogService.createLog(QueueMessageType.UNLINK, newQueueMessage(maatId));
+
+        verify(queueMessageLogRepository, times(1)).save(queueMessageCaptor.capture());
```

Comment:
Just a suggestion, the default value for times(1) is one so we don't have to explicitly add this param (times).

Reply:
Thanks, I can take it off.